### PR TITLE
add websocket support

### PR DESCRIPTION
**Pull request by stephen**
This PR:
- fixes broken fsnotify references for dep
- replaces references to bitly so that dep doesn't get confused
- adds support for websocket forwarding
- adds a test for websocket forwarding

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -89,10 +91,36 @@ func (u *UpstreamProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+}
+
+func NewReverseProxy(target *url.URL) *ReverseProxy {
+	wsTarget := new(url.URL)
+	*wsTarget = *target
+	switch target.Scheme {
+	case "https":
+		wsTarget.Scheme = "wss"
+	case "http":
+		wsTarget.Scheme = "ws"
```
Comment (stephen) [inline oauthproxy.go:117]: this is a little weird. probably should change how the url resolution works, but they need the same handler so i'm going to punt.

**Comment by berfarah** [comment]
This seems like it should've needed a review to go through.

File: oauthproxy.go (hunk 1 above)
Comment (berfarah) [reply oauthproxy.go:117]: Are these analogous? As in, wss means encrypted ws?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: go
 go:
   - 1.8.x
```
Comment (berfarah) [inline .travis.yml:3]: Should we be testing the go version we actually use as well? 1.10?

File: oauthproxy.go (hunk 1 above)
Comment (stephen) [reply oauthproxy.go:117]: yes, https://www.w3.org/TR/websockets/#parsing-websocket-urls

File: .travis.yml (hunk 2 above)
Comment (stephen) [reply .travis.yml:3]: yes, i'll follow up to do this. originally, i wanted to run this against 1.8.x and use it, but i had no easy way of compiling to 1.8.x.

**Pull request merged**